### PR TITLE
Pgadmin4: fix build. Also fix flask-security-too

### PR DESCRIPTION
**Pull request by gador**
## Description of changes

In the recent major Flask update `flask-security-too` has been updated. Unfortunately it doesn't build as is and needs fixing (including the removal of `flask-mongoengine`). 

Also, `pgadmin4` doesn't build, since it isn't yet compatible with the newer Flask ecosystem. This PR includes the `flask-security-too` and `pgadmin4` fixes (It doesn't make sense for two separate PRs because `pgadmin4` relies on `flask-security-too`).  

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 262702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>pgadmin4</li>
<li>pgadmin4-desktopmode</li>
<li>pgadmin4-desktopmode.dist</li>
<li>pgadmin4.dist</li>
<li>python310Packages.flask-security-too</li>
<li>python310Packages.flask-security-too.dist</li>
<li>python311Packages.flask-security-too</li>
<li>python311Packages.flask-security-too.dist</li>
</ul>
</details>

**Review by bendlas — APPROVED** [review]
pgadmin is currently broken on unstable, this fixes it

File: pkgs/development/python-modules/flask-security-too/default.nix [hunk 1]
```diff
@@ -47,7 +48,7 @@
 buildPythonPackage rec {
   pname = "flask-security-too";
   version = "5.3.0";
-  format = "setuptools";
+  format = "pyproject";
```
Comment (mweinelt) [inline pkgs/development/python-modules/flask-security-too/default.nix:51]:
```suggestion
  pyproject = true;
```

**Review by mweinelt — CHANGES_REQUESTED** [review]
Please add the `build-system` dependencies in `pyproject.toml` to `nativeBuildInputs`.

It's surprising that the build currently works, but that is because `qrcode` propagates `setuptools` currently. That may change at any point and then this package would break.

File: pkgs/development/python-modules/flask-security-too/default.nix [hunk 2]
```diff
@@ -92,12 +94,12 @@ buildPythonPackage rec {
     mfa = [
       cryptography
       phonenumbers
+      webauthn
```
Comment (mweinelt) [inline pkgs/development/python-modules/flask-security-too/default.nix:97]: qrcode was moved from common to mfa.

https://github.com/Flask-Middleware/flask-security/blob/5.3.0/pyproject.toml#L57

File: pkgs/development/python-modules/flask-security-too/default.nix [hunk 3]
```diff
@@ -71,6 +72,7 @@ buildPythonPackage rec {
     flask-wtf
     itsdangerous
     passlib
+    importlib-resources
   ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/flask-security-too/default.nix:76]: Update the depdendencies. We don't need the packages that flask already provides.

https://github.com/Flask-Middleware/flask-security/blob/5.3.0/pyproject.toml#L41-L51

**Review by mweinelt — DISMISSED** [review]
And maybe update to 5.3.2.

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 262702` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>pgadmin4</li>
<li>pgadmin4-desktopmode</li>
<li>pgadmin4-desktopmode.dist</li>
<li>pgadmin4.dist</li>
<li>python310Packages.flask-security-too</li>
<li>python310Packages.flask-security-too.dist</li>
<li>python311Packages.flask-security-too</li>
<li>python311Packages.flask-security-too.dist</li>
</ul>
</details>

**Comment by gador** [comment]
@mweinelt thanks for your feedback! I applied it, could you have another look?

**Comment by bendlas** [comment]
I deployed this, still runs.

since unstable is currently broken: can we merge this, and open a new PR in case we get more feedback?

**Pull request merged**